### ISSUE TITLE
Comments for "DIGImend is dead, long live DIGImend!"

### ISSUE THREAD

**Issue by spbnick** [spbnick/spbnick.github.io#9]
Please leave your comments for article "[DIGImend is dead, long live DIGImend!](http://spbnick.github.io/2017/12/03/Long-live-DIGImend.html)" here.

**Comment by asbaklm** [comment]
That was certainly an interesting post with a positive conclusion. It must be a huge relief knowing that all that time and hard work was not for nothing, and that people and companies were benefiting, even if this wasn't always apparent at the time when the project started stalling.

Any project such as this relies on motivation, on positive user feedback and on a sense of accomplishment which makes it all worth the hard time spent. The coding and engineering side is just one (although obviously the most important) part to the whole thing.

Where it becomes a struggle is to marry all the disciplines, namely the combination of coding, marketing and promotion, dealing with endless support issues and very significantly, the business side to negotiate for and attract investment. 

As your post has shown, there is commercial & vendor interest out there. The trick is attracting more business from such players and in selling and convincing them to sign on and to maintain those relationships. Business negotiations and marketing are obviously very different skills & disciplines to coding but they are quite essential. Insufficient attention to those things (the commercial side) will hamper progress.

It's a drag to have to do, particularly with all the other work that needs doing, but in order to attract business it's essential to keep reaching out to vendors to attract more work and to start building up relationships. And it needs to be done in such a way that the vendors aren't feeling that they are being spammed or annoyed. In simple terms, these companies need to be sold on an idea, on a vision. It needs to be clearly explained to them why and how they will benefit and how they can build sales momentum in new markets. The need convincing, and in order to convince them, a good business case backed up by evidence and success stories is required.

I think it would be useful if a little Linux distro (derived from a mainstream Linux) were developed (not by you, you have no time obviously) by the community which included the more popular common graphics, photo-editing & CAD packages and to have the latest drivers baked in. This could be pitched as another incentive to the manufacturers because such a distro can easily be installed on a PC type device and will work out of the box with tablets and graphics software. It may also be of use to reach out to the community at Libregraphicsworld   http://libregraphicsworld.org/   to see what kind of mutually beneficial actions can be undertaken.

One of the biggest challenges facing Linux adoption by end-users is the level of difficulty in getting the hardware & software set up and running. When you buy a Mac, or to a lesser degree, a Windows PC, most of that work has already been done for the end user. Installing software and drivers is usually fairly trivial and they can be up and working without too much drama.

The ultimate end goal with projects such as DIGImend ought to be to strive towards a similar user experience. If it is too complex and difficult, it will only keep attracting people with Linux skills and that's a very limited market. A derived distro based on a popular mainstream Linux containing tablet drivers & software and with a clearly documented hardware compatibility matrix (supported tablets)  will make it trivial for non-technical end-users to be up and running quickly. That in turn will encourage adoption by more end-users and translate into increased community uptake and sales.

Pitching such a distro at art schools, colleges etc and getting folks interested will also go a long way toward encouraging uptake. I realise that this seems a bit far removed from the actual tablet drivers but ultimately these things are all inter-related and inter-dependent.

**Comment by spbnick** [comment]
@asbaklm, thank you for your thoughtful and elaborate comment. I will have to re-read and think about this more. I have a history of cooperating with LGW, perhaps we can work something out. They have already retwitted my Patreon announce. Yes, I agree, Linux still needs to become easier for users to manage, and in HW support in particular.

Still, I think creating another distro just for tablet compatibility is too much. I think we can work something out with Ubuntu Studio and similar distros. E.g. setup or link tablet compatibility page, make a statement on compatibility, help with/participate in promotion, etc. A good goal might be getting DIGImend package into Debian. Fedora-based distros don't support out-of-tree modules, though. We also need to speed up contribution to upstream kernel.

In general, I agree we need to work on marketing and promoting the project, however boring that might sound. The business side is important.

**Comment by asbaklm** [comment]
An arrangement with Ubuntu Studio would certainly be useful but as a platform it is somewhat limited in terms of software selection and also with regard to the age of libraries and packages. (Ubuntu's packages are often pretty old.)

The newer US releases are not LTS (Long Term Support) so within a couple of months the user will have to re-install if they intend to receive system updates. It is not really an ideal situation from an end-user point of view. A user's computer shouldn't need to be re-installed every couple of months because this will serve to annoy and discourage them.

Imo the ideal solution requires (amongst other things):

- Work out of the box without requiring specialist knowledge (w. supported hardware)
- Be based on LTS releases, not releases which expire within a few months
- Be easy to maintain
- Provide a wider and more up to date selection of the more useful and popular software packages
- Be up to date with hardware support, supporting latest devices, or provide patches or driver packages which can easily be installed by end users to update hardware support.
- Also be useful for general purpose computing, web browsing, email, writing documents etc

However you are right, creating and maintaining yet another distro is not a trivial undertaking. If Ubuntu Studio's maintainers were willing to entertain suggestions for improvement that would be even better. As things currently stand I'm not convinced about US in its current form and feel that the experience is sub-optimal.

Another avenue to improve the overall marketing situation would be to create a few Youtube type videos (a mini-series) featuring artists already using compatible hardware & software and providing a summary in the video-postings about what is required to make things work. It goes without saying that a bit of thought, planning and good video-editing will be required to give the whole thing a nice feel. (One guy talking into a mic and drawing a few pictures will be too boring.) Then also provide links to the video series on blog postings etc to further cross-promote the entire ecosystem. It doesn't need to be too difficult, expensive or time-consuming to produce but it will require planning and a clear vision.

For example, there is an artist called David Revoy  https://davidrevoy.com/  who in the past has provided quite useful blogs on getting tablets working under Linux. If a few people like him would be willing to do some short videos to demonstrate what is possible, it will provide a convincing argument to show to vendors and to convince end users to make the leap from Windows or Mac to Linux.

It's a drag to do all this marketing stuff but without it, nobody outside of a small fringe group of users will even be aware that it is possible to do high quality graphic work on an Open Source Platform and that the experience need not be difficult, frustrating or painful to achieve.

**Comment by doctormo** [comment]
As an Inkscape developer I put together dkms packages for wizardpen (back when it worked) for Ubuntu users. Even though I only have wacom tablets. I've pledged a dollar to the patreon to help you along because it's important to have these drivers working.

If there is a problem compiling a plug-able dkms package for debian/ubuntu if kernel drivers can't be made in time for the Ubuntu 18.04, let me know. Maybe I can put something together and create a PPA for the packages which you could advertise on the website (depends on the patch set of course).

**Comment by spbnick** [comment]
@asbaklm, yes a video series would be great, as would marketing in general. I think getting just the base Ubuntu to support the tablets would already be very good. I have never been an artist myself and have no idea why just the basic Ubuntu doesn't work and Ubuntu Studio is necessary, but I assume there is a reason. However, if Ubuntu gets the drivers other will have an easier time to follow. Yes, David is doing a great job as a promoter of Linux and Open-Source software for artists.

**Comment by spbnick** [comment]
@doctormo, thank you for your support! I think the road to Ubuntu proper would probably lie through Debian. If I have funding, I can try pushing the drivers there. Otherwise a PPA would be much better than nothing, a step up from the .deb packages of digimend-kernel-drivers I make.

**Comment by doctormo** [comment]
The advantage of a PPA is that you get to support previous versions of Ubuntu that didn't come with the driver installed. This is important for LTS releases, since 16.04 will be supported until 2021 (https://wiki.ubuntu.com/Releases) and users may not move to newer releases.

Although weighing up how hard it is to get things into debian vs the kernel, going through that process you might just as much end up with a kernel patch in Debian rather than a dkms deb package, and those are specificity different processes between ubuntu and debian so one wouldn't necessarily lead to the other.

I used to use my PPA for testing/master branch releases too, since asking users to `apt-get install wizardpen-testing` was easier than any other instruction and they could report back issue quicker than via a normal release cycle. But this is just food for thought if you want more testing/user involvement.

**Comment by spbnick** [comment]
Right, I never really used Ubuntu and didn't think about using PPA for older releases. That would definitely be useful, if you or somebody else could take on maintaining a PPA. Thank you for the explanations.

**Comment by asbaklm** [comment]
Both angles should probably be pursued. PPA's for older distributions and getting the kernel patched so that future distributions will natively support newer tablets. Prioritising PPA's as a short term solution for the masses may be more useful. 

Getting the kernel patched is a longer term goal since the user base who will be able to derive benefit from this will initially be very small. 

I also think that trying to get changes merged into Debian and / or Ubuntu may require more work & effort than it is worth (they can be quite resistant to accepting modifications). Therefore an independently "provided as is" PPA is probably the nicest solution for less technical users.

**Comment by Alex-Esko** [comment]
Thanks for your support of many graphic-tablets.
Do you have an IBAN for donations?
PayPal would also be possible, but has much higher fees.

**Comment by spbnick** [comment]
@Alex-Esko, I don't have an IBAN specifically for donations, but I can give you my personal IBAN in an e-mail message. Please write to spbnick@gmail.com if you'd like me to do that. Otherwise I would encourage you to become a Patreon supporter. They do have their fees, though. Thank you!
